pkg/services/object/acl/v2/util.go

```diff
@@ -93,64 +93,33 @@ func originalSessionToken(header *sessionV2.RequestMetaHeader) (*sessionSDK.Obje
 	return &tok, nil
 }
 
-func getObjectIDFromRequestBody(body interface{}) (*oid.ID, error) {
-	var idV2 *refsV2.ObjectID
+// getObjectIDFromRequestBody decodes oid.ID from the common interface of the
+// object reference's holders. Returns an error if object ID is missing in the request.
+func getObjectIDFromRequestBody(body interface{ GetAddress() *refsV2.Address }) (*oid.ID, error) {
+	return decodeRequestedObject(body.GetAddress().GetObjectID(), true)
+}
 
-	switch v := body.(type) {
-	default:
-		return nil, nil
-	case interface {
-		GetObjectID() *refsV2.ObjectID
-	}:
-		idV2 = v.GetObjectID()
-	case interface {
-		GetAddress() *refsV2.Address
-	}:
-		idV2 = v.GetAddress().GetObjectID()
-	}
+// decodeRequestedObject decodes oid.ID from refsV2.ObjectID message if latter
+// is presented, otherwise returns an error iff message is required.
+func decodeRequestedObject(m *refsV2.ObjectID, required bool) (*oid.ID, error) {
+	if m == nil {
+		if required {
+			return nil, errors.New("missing object ID")
```

Comment:
It seems like we use this func in only 2 places, may be it is better to have `nil` check on the call-site and remove the second parameter?